src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -904,4 +907,183 @@ BOOST_AUTO_TEST_CASE(SerializationTest)
   CheckMatrices(prediction, xmlPrediction, textPrediction, binaryPrediction);
 }
 
+/**
+ * Test RNN with a custom layer.
+ */
+void ReberGrammarTestCustomNetwork(const size_t hiddenSize = 4,
+                                   const bool recursive = false,
+                                   const size_t averageRecursion = 3,
+                                   const size_t maxRecursion = 5,
+                                   const size_t iterations = 10)
+{
+  // Reber state transition matrix. (The last two columns are the indices to the
+  // next path).
+  arma::Mat<char> transitions;
```

Comment:
I think what we could do to shorten the test file is to write a ``GenerateReberGrammarData`` function and use that in both tests, what do you think?

Reply:
Yeah I thought of it and then thought against it because I did not want to shake things up too much. But I will give it a shot.